debugging.md

```diff
@@ -0,0 +1,107 @@
+# Debugging
+
+This document aims to collect info on the current status and future opportunities for debugging Rust.
+
+## Debugging architecture overview
+
+The compiler generates debuginfo for Rust programs (see https://github.com/rust-lang/rust/tree/master/src/librustc_trans/debuginfo).
+This is an LLVM-specific thing. LLVM then processes this through optimisations
+(with varying degrees of success) and outputs debug symbols with the library.
+These are DWARF for the Linux, Mac, and Windows/GNU toolchains, and pdb (program
+database) for Windows/visual studio (pdb support is WIP in LLVM, I believe).
+
+Debuggers consume the generated symbols. Visual Studio only works on Windows.
+LLDB works everywhere but is primarily used as a backend for XCode on MacOS. GDB
+works everywhere. GDB and LLDB are command
+line tools, but can have frontends.
+
+Debugger support can come in various forms in this architecture:
+* we could invent our own debugger and debug symbols (this is infeasible due to resourcing)
```

Comment:
I'd like to mention https://github.com/oli-obk/priroda here, since it's just a thin layer around miri. It steps through MIR and supports graphical memory browsing and some simple debugger commands (no break points yet).